### ISSUE TITLE
Class init fucntion docstring is used instead of class docstring

### ISSUE THREAD

**Issue by qumusabel** [microsoft/pyright#1666]
**Describe the bug**

**To Reproduce**
Steps to reproduce the behavior.

**Expected behavior**

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
def foo(self) -> str:
    return 3
```

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool? Which version? You can find the version of the VS Code extension by clicking on the Pyright icon in the extensions panel.

**Additional context**

**Comment by qumusabel** [comment]
I'm sorry, pressed enter accidentally. Text coming soon

**Comment by erictraut** [comment]
Thanks for submitting the issue. I think that's a reasonable perspective, but there's also an argument that a constructor docstring should take precedence. Some discussion is needed here. Since this issue is related to language server functionality and not a core type checking feature, I'm going to transfer this to the pylance repo. It will get the appropriate attention and discussion there. If we decide to make a change in this functionality, it will be in both pyright and pylance.
